pkg/oc/cli/secrets/options.go

```diff
@@ -68,6 +69,16 @@ func (o SecretOptions) Validate() error {
 		return errors.New("KubeCoreClient must be present")
 	}
 
+	// if any secret names are of the form <resource>/<name>,
+	// ensure <resource> is a secret.
+	for _, secretName := range o.SecretNames {
```

Comment:
Why isn't this using standard resource builder?

Reply:
>Why isn't this using standard resource builder?

The resource builder is [used in the `GetSecrets` method](https://github.com/openshift/origin/blob/master/pkg/oc/cli/secrets/options.go#L166). The code block here validates that a resource kind is `secret` if it is specified as `<kind>/<name>`, before getting to the builder, which ignores `NotFound` errors in order to allow  deleted secrets to still be unlinked. Without this check, if a user provides a command such as: `oc secret unlink sa_name pods/not_a_secret`, this line would create a secret with the name `pods/not_a_secret`.